### PR TITLE
Run e2e tests on MacOS in CI

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,6 +57,8 @@ jobs:
         include:
           - os: ubuntu-latest
             e2e: true
+          - os: macos-latest
```
Comment (bonomat) [inline .github/workflows/ci.yml:60]: Simpler would be to explicitly *not* run it on windows-latest

**Comment by tcharding** [comment]
Kicked bors without second approval while trying to debug bors crash.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/ci.yml:60]: Agreed but I didn't want to mess with the config too much.

Ideally, I would run them on windows too but for that we need to change how we download the blockchain nodes.

**Pull request merged**